MeshLib/MeshSubset.h

```diff
@@ -53,7 +53,24 @@ class MeshSubset
     /// \param vec_items Vector of Node pointers.
     MeshSubset(const Mesh& msh, std::vector<Node*> const& vec_items)
         : _msh(msh), _nodes(vec_items)
-    {}
+    {
+        // testing if the given nodes belong to the mesh.
+        auto node_is_part_of_mesh =
+            [mesh_nodes = _msh.getNodes()](MeshLib::Node* const& n) {
+                auto it = std::find(begin(mesh_nodes), end(mesh_nodes), n);
+                if (it == end(mesh_nodes))
+                {
+                    ERR("A node %d in mesh subset is not a part of the mesh.",
+                        n->getID());
+                    return false;
+                }
+                return true;
+            };
+        if (!std::all_of(begin(_nodes), end(_nodes), node_is_part_of_mesh))
```

Comment:
The runtime complexity of that is `length(mesh_nodes) * length(mesh_subset)` isn't that terribly slow?

Reply:
You are right, although it is not that slow that it would have drawn my attention to it.
I'll update for more optimized version in this PR. The special case (also very fast to check) is the whole mesh subset, where the nodes are the same as the mesh's nodes.

Reply:
Improving it, tested with GWF/square_1e5_neumann.prj

ufz/master time is 2.88s
the unsorted vector naive implementation runs in 4.95s
the sorted vector with binary search runs in 2.90s
and checking for the vector addresses before expensive search is indistinguishable from the ufz/master.

Taking the last one :)